monte_carlo.cpp

```diff
@@ -0,0 +1,321 @@
+  
+                // if sur_side is in box_lim_bound
+                if (std::find(box_lim_bound.begin(),
+                            box_lim_bound.end(),sur_side)
+                        != box_lim_bound.end()) {
+
+                    // if the neutron is reflected
+                    if (bounds.getSurfaceType(axis, side) == 1) {
+                        neutron.reflect(axis);
+
+                        // place neutron on boundary to eliminate 
+                        //    roundoff error
+                        bound_val = bounds.getSurfaceCoord(axis, side);
+                        neutron.setPosition(axis, bound_val);
+                    }
+
+                    // if the neutron escapes
+                    if (bounds.getSurfaceType(axis, side) == 0) {
+                        neutron.kill();
+                        neutron_distance = tempd;
+                        tallies[LEAKS].add(1);
+                    }
+                }
+            }
+ 
+            // shorten neutron distance to collision
+            neutron_distance -= tempd;
+
+            // get new neutron cell
+            if (neutron_distance != 0.0) {
+                neutron_direction = neutron.getDirectionVector();
+                cell = mesh.getCell(neutron_position, neutron_direction);
+
+                // nudge neutron
+                neutron.move(TINY_MOVE);
+                neutron_position = neutron.getPositionVector();
+                if (mesh.positionInBounds(neutron_position)) {
+                    cell = mesh.getCell(neutron_position, neutron_direction);
+                }
+                neutron.move(-TINY_MOVE);
+                neutron.setCell(cell);
+            }
+        }
+
+        // check interaction
+        if (neutron.alive()) {
+            cell_mat = mesh.getMaterial(cell);
+
+            // sample what the interaction will be
+            neutron_interaction = sampleInteraction(cell_mat, group);
+
+            // scattering event
+            if (neutron_interaction ==0) {
+
+                // sample scattered direction
+                theta = samplePolarAngle();
+                phi = sampleAzimuthalAngle();
+
+                // sample new energy group
+                temp_sigma_s = cell_mat->getSigmaS();
+                new_group = sampleScatteredGroup(temp_sigma_s, group);
+
+                // set new group
+                neutron.setGroup(new_group);
+
+                // set new direction
+                neutron.setDirection(theta, phi);
+            }
+
+            // absorption event
+            else {
+
+                // tally absorption
+                tallies[ABSORPTIONS].add(1);
+
+                // sample for fission event
+                group = neutron.getGroup();
+                cell = neutron.getCell();
+                cell_mat = mesh.getMaterial(cell);
+                neutron_position = neutron.getPositionVector();
+                if (sampleFission(cell_mat, group) == 1) {
```

Comment:
I would add a comment before the conditional to make this code look a bit cleaner